### PR TITLE
[SyntaxParse] Re-apply move-only ParsedRawSyntaxNode

### PR DESCRIPTION
**Pull request by rintaro**
Revert #27203 with a fix.

Locally confirmed this successfully compiles stdlib with asan+ubsan in both debug and release build.

rdar://problem/55421369

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test ASAN

**Comment by rintaro** [comment]
@swift-ci Please ASAN test

File: lib/Parse/SyntaxParsingContext.cpp [hunk 1]
```diff
@@ -163,7 +164,8 @@ const SyntaxParsingContext *SyntaxParsingContext::getRoot() const {
 }
 
 ParsedTokenSyntax SyntaxParsingContext::popToken() {
-  return popIf<ParsedTokenSyntax>().getValue();
+  auto tok = popIf<ParsedTokenSyntax>();
+  return std::move(tok.getValue());
```
Comment (rintaro) [inline lib/Parse/SyntaxParsingContext.cpp]: In https://github.com/apple/swift/pull/27190, this was:
```cpp
ParsedTokenSyntax &&SyntaxParsingContext::popToken() {
  return std::move(popIf<ParsedTokenSyntax>().getValue());
}
```
This returns a reference to the temporary local value of `popIf<ParsedTokenSyntax>()`.

Comment (rintaro) [reply lib/Parse/SyntaxParsingContext.cpp]: @compnerd Could you check if this change doesn't break Windows build?

**Comment by rintaro** [comment]
https://ci.swift.org/job/swift-PR-osx-ASAN-test/96/ Infrastructure issue?

**Comment by rintaro** [comment]
@swift-ci Please ASAN test

**Comment by CodaFi** [comment]
The ASAN bot builds in release with debug info. It shouldn't catch this.

**Comment by rintaro** [comment]
@CodaFi  As I noted in the description, I confirmed this change successfully compiles stdlib with asan+ubsan in both debug and release build.

@shahmishal Is there a way to invoke pull request CI with debug+ASAN ?

**Comment by jrose-apple** [comment]
@swift-ci Please test Windows

**Comment by rintaro** [comment]
preset=buildbot_incremental,tools=DA,stdlib=DA,build
@swift-ci Please test with preset macOS Platform

**Pull request merged**